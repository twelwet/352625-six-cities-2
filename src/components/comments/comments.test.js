import React from "react";
import renderer from "react-test-renderer";
import Comments from "./comments.jsx";

const props = {
  comments: [
    {
      comment: `some comment text`,
      date: `2019-11-25T12:13:30.711Z`,
      id: 34,
      rating: 3.6,
      user: {
        [`avatar_url`]: `filepath`,
        id: 4,
        [`is_pro`]: true,
        name: `Barbara`
      }
    },
    {
      comment: `other comment text`,
      date: `2019-11-25T12:15:45.711Z`,
      id: 32,
      rating: 2.9,
      user: {
        [`avatar_url`]: `filepath`,
        id: 5,
        [`is_pro`]: false,
        name: `Andrew`
      }
    }
  ]
};

it(`Comments correctly renders after relaunch`, () => {

  const tree = renderer
    .create(<Comments {...props} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
