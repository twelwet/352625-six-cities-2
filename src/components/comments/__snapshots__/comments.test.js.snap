// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments correctly renders after relaunch 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="filepath"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Andrew
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "58%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          other comment text
        </p>
        <time
          className="reviews__time"
          dateTime="2019-11-25"
        >
          November 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="filepath"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Barbara
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "72%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          some comment text
        </p>
        <time
          className="reviews__time"
          dateTime="2019-11-25"
        >
          November 2019
        </time>
      </div>
    </li>
  </ul>
  
</section>
`;
